src/Bitcoin/TransactionBuilder.cpp

```diff
@@ -61,29 +61,46 @@ int64_t estimateSegwitFee(FeeCalculator& feeCalculator, const TransactionPlan& p
     return fee;
 }
 
+int64_t TransactionBuilder::getTotalAmountFromInput(const Bitcoin::Proto::SigningInput& input) {
+    int64_t sum = input.amount();
+    for (auto i = 0; i < input.extra_outputs_size(); ++i) {
+        sum += input.extra_outputs(i).amount();
+    }
+    return sum;
+}
+
 TransactionPlan TransactionBuilder::plan(const Bitcoin::Proto::SigningInput& input) {
     auto plan = TransactionPlan();
-    plan.amount = input.amount();
+    plan.amount = getTotalAmountFromInput(input);
 
-    auto output_size = 2;
     auto& feeCalculator = getFeeCalculator(static_cast<TWCoinType>(input.coin_type()));
```

Comment:
Can be to add const? this can give you some optimization. Add getFeeCalculator return const ref.

Reply:
Const has to be propagated to UnspentSelector.feeCalculator field, but it is possible, thanks.